lib/Controller/DocumentController.php

```diff
@@ -469,6 +469,23 @@ private function handleDocIndex($fileId, $shareToken, $currentUser) {
 		return $docIndex;
 	}
 
+	/**
+	 * Accessed from external-apps such as new owncloud web front-end
+	 * It returns the information to load the document from the fileId.
+	 * @NoAdminRequired
+	 * @CORS
+	 * @NoCSRFRequired
+	 */
+	public function getDocumentIndex($fileId) {
+		try {
+			$docRetVal = $this->handleDocIndex($fileId, null, \OC_User::getUser());
```

Comment:
well, $this->uid was my first choice, but unfortunately, there's a problem with it and it becomes null on CORS requests. Only \OC_User::getUser() works. Otherwise we get bad request.